WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListFragment.java

```diff
@@ -661,17 +743,55 @@ public void onPostUploaded(OnPostUploaded event) {
         }
     }
 
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPostsSearched(OnPostsSearched event) {
+        mIsSearching = false;
+        hideLoadMoreProgress();
+
+        if (event.isError()) {
+            AppLog.w(AppLog.T.POSTS, "Error searching posts: " + event.error.message);
+            ToastUtils.showToast(getActivity(), "Error searching posts: " + event.error.type);
+        } else {
+            if (event.searchResults != null && !ListUtils.isEmpty(event.searchResults.getPosts())) {
+                hideEmptyView();
+                mSearchResults.addAll(event.searchResults.getPosts());
+                setLocalPostIdsOnSearchResults();
+                getPostListAdapter().setPostList(mSearchResults);
+            } else {
+                // search returned no results
+                getPostListAdapter().setPostList(mSearchResults);
+                updateEmptyView(EmptyViewMessageType.NO_CONTENT);
+            }
+            mCanSearchMore = event.canLoadMore;
+        }
+    }
+
     /*
      * Media info for a post's featured image has been downloaded, tell
      * the adapter so it can show the featured image now that we have its URL
      */
     @SuppressWarnings("unused")
     @Subscribe(threadMode = ThreadMode.MAIN)
     public void onMediaChanged(MediaStore.OnMediaChanged event) {
-        if (isAdded() && !event.isError() && mPostsListAdapter != null) {
+        if (isAdded() && !event.isError()) {
             if (event.mediaList != null && event.mediaList.size() > 0) {
                 MediaModel mediaModel = event.mediaList.get(0);
-                mPostsListAdapter.mediaChanged(mediaModel);
+                getPostListAdapter().mediaChanged(mediaModel);
+            }
+        }
+    }
+
+    private void setLocalPostIdsOnSearchResults() {
```

Comment:
Opened wordpress-mobile/WordPress-FluxC-Android#476 to take care of updating the local ID's.